src/main/java/io/kestra/storage/azure/AzureStorage.java

```diff
@@ -49,10 +50,32 @@ public InputStream get(String tenantId, URI uri) throws IOException {
 
             return blobClient.openInputStream();
         } catch (BlobStorageException e) {
-            throw new IOException(e);
+            throw reThrowBlobStorageException(uri, e);
         }
     }
 
+    @Override
+    public List<FileAttributes> list(String tenantId, URI uri) throws IOException {
+        String path = getPath(tenantId, uri);
+        String prefix = (path.endsWith("/")) ? path : path + "/";
+
+        ListBlobsOptions listBlobsOptions = new ListBlobsOptions()
+            .setPrefix(prefix)
+            .setDetails(new BlobListDetails().setRetrieveDeletedBlobs(false).setRetrieveSnapshots(false));
+
+        return this.client().listBlobsByHierarchy("/", listBlobsOptions, null).stream()
+            .map(BlobItem::getName)
+            .filter(key -> {
+                key = "/" + key;
+                key = key.substring(prefix.length());
+                // Remove recursive result and requested dir
+                return !key.isEmpty() && !Objects.equals(key, prefix) && new File(key).getParent() == null
```

Comment:
I just use the file to check if there is a parent more easily I could also check of a "/" neither in the end or the begining maybe this would help the readability ?

Reply:
I think we already check using the last "/" somewhere so better use this than create a file wich make a reader think you will do file opertaions